_posts/2020/2020-01-10-gasper-part-1.md

```diff
@@ -0,0 +1,89 @@
+---
+ layout: post
+ title: Gasper - Your cloud in a binary (Part 1)
+ excerpt: Gasper is an intelligent Platform as a Service (PaaS) that gets rid of all the hassles and makes deploying applications amazingly simple (and fast too!).
+ author:
+   name: Supratik Das
+   link: https://github.com/supra08
+   bio: Developer, SDSLabs
+   image: supratik.jpg
+---
+
+*This blog is Part 1 of a series. It gives an idea of Gasper’s architecture, technical stack, and capabilities. The subsequent parts cover the detailed discussion of its various components.*
+
+## Introduction
+Providing simplicity without letting go of the necessities in deployments has always been a bottleneck for PaaS providers. Imagine you have a couple of Bare-Metal servers or Virtual Machines at your disposal (collectively called here as nodes) and you wanted to setup an application hosting service, such that the load is evenly distributed among all nodes.
+
+#### The dilemma
+Let’s face it; we already have a couple of choices.
+
+*Manually Setup Everything:* \\
+Manually decide which application goes to which node, then use ssh/telnet to manually setup all of your applications one by one. 
+
+*Kubernetes at the rescue:* \\
+Kubernetes takes away the load distribution problem with its intelligent clustering. But then again, it requires a lot (and yes a lot!) of configurations for each application. Besides, the pipeline for setting up the app still needs to be implemented.
+
+#### The solution
+Well, there you have it. Gasper addresses all of the above problems and makes deploying applications just a breeze. The platform is modeled as a REST API, where the user just needs to provide the Github repository’s URL, build commands and some optional configurations, click the Create button and voila, your application is deployed!.
+
+## Sounds cool? Let's look at the amazing tale under the hood
+
+Gasper is written entirely in Go - [the language of the Cloud](https://thenewstack.io/go-the-programming-language-of-the-cloud/). It essentially has a microservices-based architecture. If you don’t know already, we have multiple software design paradigms. One is the monolithic architecture, the whole application as a single unit with multiple modules. These applications lose flexibility easily and are difficult to scale. Next, we have the microservices architecture. Each module is a separate unit and interacts only with APIs. Technically, we say that we have decoupled the modules. This can be scaled easily, and each module can be designed separately.
```

Comment:
architecture ki alag se subheading bana. Start off with telling about different sofware architectures, and move on to why we picked microservices